File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/HomeworkActAdm/CreateHomeworkFragAdm.java
package com.example.adminapp.HomeworkActAdm;

import static android.app.Activity.RESULT_OK;

import android.annotation.SuppressLint;
import android.app.DatePickerDialog;
import android.app.TimePickerDialog;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.TextView;
import android.widget.TimePicker;
import android.widget.Toast;

import com.example.adminapp.R;
import com.google.android.material.datepicker.MaterialStyledDatePickerDialog;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;


public class CreateHomeworkFragAdm extends Fragment {
    final Calendar myCal = Calendar.getInstance();
    private TextView startDateTVCreateHomeworkFragAdm,endDateTVCreateHomeworkFragAdm
            ,startTimeTVCreateHomeworkFragAdm,endTimeTVCreateHomeworkFragAdm,attachmentTvCreateHomeworkFragAdm;
   private String uStartTime,uEndTime;
   private String uEndDate,uStartDate;
    private int Hour,Minute;
    private Date endDateObj,startDateObj;
    private Button postHomeWoreBtnCreateHomeworkFragAdm;
    private Boolean startDateTvClicked=false,endDateTvClicked=false,startTimeTvClicked=false,endTimeTvClicked=false;

    public CreateHomeworkFragAdm() {
        // Required empty public constructor
    }



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = LayoutInflater.from(getContext()).inflate(
                R.layout.fragment_create_homework_frag_adm,container,false);
        startDateTVCreateHomeworkFragAdm = view.findViewById(R.id.startDateTVCreateHomeworkFragAdm);
        endDateTVCreateHomeworkFragAdm = view.findViewById(R.id.endDateTVCreateHomeworkFragAdm);
        postHomeWoreBtnCreateHomeworkFragAdm = view.findViewById(R.id.postHomeWoreBtnCreateHomeworkFragAdm);

        startTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.startTimeTVCreateHomeworkFragAdm);
        endTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.endTimeTVCreateHomeworkFragAdm);

        startTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.startTimeTVCreateHomeworkFragAdm);
        endTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.endTimeTVCreateHomeworkFragAdm);


        DatePickerDialog.OnDateSetListener startDate =  new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view, int year, int month, int day) {
                myCal.set(Calendar.YEAR,year);
                myCal.set(Calendar.MONTH,month);
                myCal.set(Calendar.DAY_OF_MONTH,day);
                startDateUpdateLabel();
            }
        };
        DatePickerDialog.OnDateSetListener endDate =  new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view, int year, int month, int day) {
                myCal.set(Calendar.YEAR,year);
                myCal.set(Calendar.MONTH,month);
                myCal.set(Calendar.DAY_OF_MONTH,day);
                endDateUpdateLabel();
            }
        };

        startDateTVCreateHomeworkFragAdm.setOnClickListener(new View.OnClickListener() {
            @SuppressLint("RestrictedApi")
            @Override
            public void onClick(View view) {
                startDateTvClicked = true;
                new MaterialStyledDatePickerDialog(view.getContext(),startDate,
                        myCal.get(Calendar.YEAR),myCal.get(Calendar.MONTH),myCal.get(Calendar.DAY_OF_MONTH)).show();
            }
        });
        endDateTVCreateHomeworkFragAdm.setOnClickListener(new View.OnClickListener() {
            @SuppressLint("RestrictedApi")
            @Override
            public void onClick(View view) {
                endDateTvClicked = true;
                new MaterialStyledDatePickerDialog(view.getContext(),endDate,
                        myCal.get(Calendar.YEAR),myCal.get(Calendar.MONTH),myCal.get(Calendar.DAY_OF_MONTH)).show();
            }
        });

        startTimeTVCreateHomeworkFragAdm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startTimeTvClicked = true;
                TimePickerDialog.OnTimeSetListener onTimeSetListener = new TimePickerDialog.OnTimeSetListener() {
                    @Override
                    public void onTimeSet(TimePicker timePicker, int selectedHour, int selectedMinute) {
                        String AM_PM ;
                        Hour = selectedHour;
                        Minute = selectedMinute;
                        if(selectedHour < 12) {
                            AM_PM = "AM";
                        } else {
                            AM_PM = "PM";
                        }
                startTimeTVCreateHomeworkFragAdm.setText(selectedHour + ":" + selectedMinute + " " + AM_PM);
                uStartTime = startTimeTVCreateHomeworkFragAdm.getText().toString();
                    }
                };
                TimePickerDialog timePickerDialog = new TimePickerDialog(getContext(),onTimeSetListener,Hour,Minute,false);
                timePickerDialog.setTitle("Select Time");
                timePickerDialog.show();
            }

        });
        endTimeTVCreateHomeworkFragAdm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                endTimeTvClicked = true;
                TimePickerDialog.OnTimeSetListener onTimeSetListener = new TimePickerDialog.OnTimeSetListener() {
                    @Override
                    public void onTimeSet(TimePicker timePicker, int selectedHour, int selectedMinute) {
                        String AM_PM ;
                        Hour = selectedHour;
                        Minute = selectedMinute;
                        if(selectedHour < 12) {
                            AM_PM = "AM";
                        } else {
                            AM_PM = "PM";
                        }
                endTimeTVCreateHomeworkFragAdm.setText(selectedHour + ":" + selectedMinute + " " + AM_PM);
                uEndTime = endTimeTVCreateHomeworkFragAdm.getText().toString();
                    }
                };
                TimePickerDialog timePickerDialog = new TimePickerDialog(getContext(),onTimeSetListener,Hour,Minute,false);
                timePickerDialog.setTitle("Select Time");
                timePickerDialog.show();
            }
        });

        attachmentTvCreateHomeworkFragAdm = view.findViewById(R.id.attachmentTvCreateHomeworkFragAdm);
        attachmentTvCreateHomeworkFragAdm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                int PICKFILE_RESULT_CODE=1;
                Intent acessFilesIntent = new Intent(Intent.ACTION_GET_CONTENT);
                acessFilesIntent.setType("*/*");
                startActivityForResult(acessFilesIntent,PICKFILE_RESULT_CODE);

            }
        });


        postHomeWoreBtnCreateHomeworkFragAdm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                postButtonClicked();
            }
        });
        return view;
    }//End OnCreate:



    @Override
    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        switch (requestCode){
            case 1:
                if(resultCode==RESULT_OK){
                    String filePath = data.getData().getPath();
                    attachmentTvCreateHomeworkFragAdm.setText(filePath);
                }
        }
    }

    private void endDateUpdateLabel() {
        String myFormat="dd/MM/yy";
        SimpleDateFormat endDateFormat=new SimpleDateFormat(myFormat, Locale.US);
        uEndDate = endDateFormat.format(myCal.getTime());
        endDateTVCreateHomeworkFragAdm.setText(uEndDate);
        try {
            endDateObj=new SimpleDateFormat("dd/MM/yyyy").parse(uEndDate);
        } catch (ParseException e) {
            e.printStackTrace();
        }
    }

    private void startDateUpdateLabel() {
        String myFormat="dd/MM/yy";
        SimpleDateFormat startDateFormat=new SimpleDateFormat(myFormat);
        uStartDate = startDateFormat.format(myCal.getTime());
        startDateTVCreateHomeworkFragAdm.setText(uStartDate);
        try {
            startDateObj=new SimpleDateFormat("dd/MM/yyyy").parse(uStartDate);
        } catch (ParseException e) {
            e.printStackTrace();
        }
    }


    /*use compareTo() for dates compare */
    private void postButtonClicked(){
        //TODO : sendDataTOApi() & Toast sahi krna hai
        validateNverify();
    }

    private void validateNverify(){
        //check click hua h date n time par
        if(startDateTvClicked && endDateTvClicked && startTimeTvClicked && endTimeTvClicked) {
            //tv are clicked atleast
            if(uStartDate.equals(uEndDate) && uStartTime.equals(uEndTime)){
                //start n end date same dala hai n time diff h
                Toast.makeText(getContext(), "Since Dates r equal Time cannot be same", Toast.LENGTH_SHORT).show();
            }else if(uStartDate.equals(uEndDate) && !uStartTime.equals(uEndTime)){
                Toast.makeText(getContext(), "2nd executed", Toast.LENGTH_SHORT).show();
            } else if(startDateObj.compareTo(endDateObj)<0){
                //agar start date chota h end se to time se kya hi mtlb
                Toast.makeText(getContext(), "start date less than end ok ideal case", Toast.LENGTH_SHORT).show();
            }else if(startDateObj.compareTo(endDateObj)>0){
                //mtlb start date bara h end se OR YEH ALLOW NAHI HAI
                Toast.makeText(getContext(), "start great then last no valid", Toast.LENGTH_SHORT).show();
            }
        }
        else{
            Toast.makeText(getContext(), "Last else part executed", Toast.LENGTH_SHORT).show();
        }
    }



}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/HomeworkActAdm/CreateHomeworkFragAdm.kt
package com.example.adminapp.HomeworkActAdm

import android.app.Activity
import android.app.DatePickerDialog
import android.app.TimePickerDialog
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.example.adminapp.R
import com.google.android.material.datepicker.MaterialStyledDatePickerDialog
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.Locale

class CreateHomeworkFragAdm : Fragment() {
    val myCal = Calendar.getInstance()
    private var startDateTVCreateHomeworkFragAdm: TextView? = null
    private var endDateTVCreateHomeworkFragAdm: TextView? = null
    private var startTimeTVCreateHomeworkFragAdm: TextView? = null
    private var endTimeTVCreateHomeworkFragAdm: TextView? = null
    private var attachmentTvCreateHomeworkFragAdm: TextView? = null
    private var uStartTime: String? = null
    private var uEndTime: String? = null
    private var uEndDate: String? = null
    private var uStartDate: String? = null
    private var Hour = 0
    private var Minute = 0
    private var endDateObj: Date? = null
    private var startDateObj: Date? = null
    private var postHomeWoreBtnCreateHomeworkFragAdm: Button? = null
    private var startDateTvClicked = false
    private var endDateTvClicked = false
    private var startTimeTvClicked = false
    private var endTimeTvClicked = false
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = LayoutInflater.from(context).inflate(
            R.layout.fragment_create_homework_frag_adm, container, false
        )
        startDateTVCreateHomeworkFragAdm = view.findViewById(R.id.startDateTVCreateHomeworkFragAdm)
        endDateTVCreateHomeworkFragAdm = view.findViewById(R.id.endDateTVCreateHomeworkFragAdm)
        postHomeWoreBtnCreateHomeworkFragAdm =
            view.findViewById(R.id.postHomeWoreBtnCreateHomeworkFragAdm)
        startTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.startTimeTVCreateHomeworkFragAdm)
        endTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.endTimeTVCreateHomeworkFragAdm)
        startTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.startTimeTVCreateHomeworkFragAdm)
        endTimeTVCreateHomeworkFragAdm = view.findViewById(R.id.endTimeTVCreateHomeworkFragAdm)
        val startDate = DatePickerDialog.OnDateSetListener { view, year, month, day ->
            myCal[Calendar.YEAR] = year
            myCal[Calendar.MONTH] = month
            myCal[Calendar.DAY_OF_MONTH] = day
            startDateUpdateLabel()
        }
        val endDate = DatePickerDialog.OnDateSetListener { view, year, month, day ->
            myCal[Calendar.YEAR] = year
            myCal[Calendar.MONTH] = month
            myCal[Calendar.DAY_OF_MONTH] = day
            endDateUpdateLabel()
        }
        startDateTVCreateHomeworkFragAdm?.setOnClickListener(View.OnClickListener { view ->
            startDateTvClicked = true
//            MaterialStyledDatePickerDialog(
//                view.context, startDate,
//                myCal[Calendar.YEAR], myCal[Calendar.MONTH], myCal[Calendar.DAY_OF_MONTH]
//            ).show()

            val c = Calendar.getInstance()
            val year = c.get(Calendar.YEAR)
            val month = c.get(Calendar.MONTH)
            val day = c.get(Calendar.DAY_OF_MONTH)

            // Date Picker Dialog
            val datePickerDialog = DatePickerDialog(view.context,
                { view, year, monthOfYear, dayOfMonth ->
                    // This is where you handle the date the user has selected
                    val selectedDate = "$dayOfMonth/${monthOfYear + 1}/$year"
                    // Example: Show the selected date
//                    Toast.makeText(view, "Selected Date: $selectedDate", Toast.LENGTH_LONG).show()
                }, year, month, day)
            datePickerDialog.show()

        })
        endDateTVCreateHomeworkFragAdm?.setOnClickListener(View.OnClickListener { view ->
            endDateTvClicked = true
//            MaterialStyledDatePickerDialog(
//                view.context, endDate,
//                myCal[Calendar.YEAR], myCal[Calendar.MONTH], myCal[Calendar.DAY_OF_MONTH]
//            ).show()

            val c = Calendar.getInstance()
            val year = c.get(Calendar.YEAR)
            val month = c.get(Calendar.MONTH)
            val day = c.get(Calendar.DAY_OF_MONTH)

            // Date Picker Dialog
            val datePickerDialog = DatePickerDialog(view.context,
                { view, year, monthOfYear, dayOfMonth ->
                    // This is where you handle the date the user has selected
                    val selectedDate = "$dayOfMonth/${monthOfYear + 1}/$year"
                    // Example: Show the selected date
//                    Toast.makeText(view, "Selected Date: $selectedDate", Toast.LENGTH_LONG).show()
                }, year, month, day)
            datePickerDialog.show()

        })
        startTimeTVCreateHomeworkFragAdm?.setOnClickListener(View.OnClickListener {
            startTimeTvClicked = true
            val onTimeSetListener =
                TimePickerDialog.OnTimeSetListener { timePicker, selectedHour, selectedMinute ->
                    val AM_PM: String
                    Hour = selectedHour
                    Minute = selectedMinute
                    AM_PM = if (selectedHour < 12) {
                        "AM"
                    } else {
                        "PM"
                    }
                    startTimeTVCreateHomeworkFragAdm?.setText("$selectedHour:$selectedMinute $AM_PM")
                    uStartTime = startTimeTVCreateHomeworkFragAdm?.getText().toString()
                }
            val timePickerDialog = TimePickerDialog(context, onTimeSetListener, Hour, Minute, false)
            timePickerDialog.setTitle("Select Time")
            timePickerDialog.show()
        })
        endTimeTVCreateHomeworkFragAdm?.setOnClickListener(View.OnClickListener {
            endTimeTvClicked = true
            val onTimeSetListener =
                TimePickerDialog.OnTimeSetListener { timePicker, selectedHour, selectedMinute ->
                    val AM_PM: String
                    Hour = selectedHour
                    Minute = selectedMinute
                    AM_PM = if (selectedHour < 12) {
                        "AM"
                    } else {
                        "PM"
                    }
                    endTimeTVCreateHomeworkFragAdm?.setText("$selectedHour:$selectedMinute $AM_PM")
                    uEndTime = endTimeTVCreateHomeworkFragAdm?.getText().toString()
                }
            val timePickerDialog = TimePickerDialog(context, onTimeSetListener, Hour, Minute, false)
            timePickerDialog.setTitle("Select Time")
            timePickerDialog.show()
        })
        attachmentTvCreateHomeworkFragAdm =
            view.findViewById(R.id.attachmentTvCreateHomeworkFragAdm)
        attachmentTvCreateHomeworkFragAdm?.setOnClickListener(View.OnClickListener {
            val PICKFILE_RESULT_CODE = 1
            val acessFilesIntent = Intent(Intent.ACTION_GET_CONTENT)
            acessFilesIntent.setType("*/*")
            startActivityForResult(acessFilesIntent, PICKFILE_RESULT_CODE)
        })
        postHomeWoreBtnCreateHomeworkFragAdm?.setOnClickListener(View.OnClickListener { postButtonClicked() })
        return view
    } //End OnCreate:

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        when (requestCode) {
            1 -> if (resultCode == Activity.RESULT_OK) {
                val filePath = data!!.data!!.path
                attachmentTvCreateHomeworkFragAdm!!.text = filePath
            }
        }
    }

    private fun endDateUpdateLabel() {
        val myFormat = "dd/MM/yy"
        val endDateFormat = SimpleDateFormat(myFormat, Locale.US)
        uEndDate = endDateFormat.format(myCal.time)
        endDateTVCreateHomeworkFragAdm!!.text = uEndDate
        try {
            endDateObj = SimpleDateFormat("dd/MM/yyyy").parse(uEndDate)
        } catch (e: ParseException) {
            e.printStackTrace()
        }
    }

    private fun startDateUpdateLabel() {
        val myFormat = "dd/MM/yy"
        val startDateFormat = SimpleDateFormat(myFormat)
        uStartDate = startDateFormat.format(myCal.time)
        startDateTVCreateHomeworkFragAdm!!.text = uStartDate
        try {
            startDateObj = SimpleDateFormat("dd/MM/yyyy").parse(uStartDate)
        } catch (e: ParseException) {
            e.printStackTrace()
        }
    }

    /*use compareTo() for dates compare */
    private fun postButtonClicked() {
        //TODO : sendDataTOApi() & Toast sahi krna hai
        validateNverify()
    }

    private fun validateNverify() {
        //check click hua h date n time par
        if (startDateTvClicked && endDateTvClicked && startTimeTvClicked && endTimeTvClicked) {
            //tv are clicked atleast
            if (uStartDate == uEndDate && uStartTime == uEndTime) {
                //start n end date same dala hai n time diff h
                Toast.makeText(
                    context,
                    "Since Dates r equal Time cannot be same",
                    Toast.LENGTH_SHORT
                ).show()
            } else if (uStartDate == uEndDate && uStartTime != uEndTime) {
                Toast.makeText(context, "2nd executed", Toast.LENGTH_SHORT).show()
            } else if (startDateObj!!.compareTo(endDateObj) < 0) {
                //agar start date chota h end se to time se kya hi mtlb
                Toast.makeText(
                    context,
                    "start date less than end ok ideal case",
                    Toast.LENGTH_SHORT
                ).show()
            } else if (startDateObj!!.compareTo(endDateObj) > 0) {
                //mtlb start date bara h end se OR YEH ALLOW NAHI HAI
                Toast.makeText(context, "start great then last no valid", Toast.LENGTH_SHORT).show()
            }
        } else {
            Toast.makeText(context, "Last else part executed", Toast.LENGTH_SHORT).show()
        }
    }
}